java/README.md

```diff
@@ -1,3 +1,14 @@
-# Azure Cosmos sample for Java for Bulk ingestion and capabilities on Graph API accounts
+# Azure Cosmos sample for Java for Bulk ingestion and capabilities on Gremlin API accounts
 
-TBD
\ No newline at end of file
+This sample uses the Java SDK v4.x.x for Cosmos Db to execute bulk operations against a Cosmos Database that has been
+configured to use the Gremlin API.
+
+This sample is useful when hitting performance issues when uploading Vertices and Edges in bulk.
+
+There are two primary components of this sample:
+
+[GraphBulkExecutor](src/main/java/com/azure/graph/bulk/impl/): Contains the source code for the reusable code that can
```

Comment:
Contains the source code of the core bulk execution functionality for Gremlin API. 


Saying this should be enough - as the 'sample' part explains what one can do with that.